File: 8-string-to-integer-atoi/string-to-integer-atoi.py
class Solution(object):
    def myAtoi(self, s):
        """
        :type s: str
        :rtype: int
        """
        s=s.strip()
        if not s:
            return 0
        ch=False
        num=0
        if s[0]=='-':
            ch=True
        elif s[0]=='+':
            ch=False
        elif not s[0].isnumeric():
            return 0
        else:
            num=ord(s[0])-ord('0')
        for i in range(1,len(s)):
            if s[i].isnumeric():
                num*=10
                num+=ord(s[i])-ord('0')
                if not ch and num>=2147483647:
                    return 2147483647
                if ch and num>=2147483648:
                    return -2147483648
            else:
                break
        if ch:
            num*=-1
        return num

File: 8-string-to-integer-atoi/string-to-integer-atoi.cpp
class Solution {
public:
    int myAtoi(string s) {
        int n = s.length();
        if(n==0)return 0;
        int num=0,i=0,sign=1;
        while(s[i]==' '){
            i++;
        }

        if(i<n && s[i]=='+'||s[i]=='-'){
            sign=s[i]=='+'?1:-1;
            i++;
        }

        while(i<n && isdigit(s[i])){
            if(num>INT_MAX/10 || num==INT_MAX/10 && s[i]>'7') return sign==1?INT_MAX:INT_MIN;
            num=num*10+(s[i]-'0');
            i++;            
        }
        return num*sign;
        
    }
};